modin/pandas/indexing.py

```diff
@@ -633,6 +633,25 @@ def __getitem__(self, key):
         if self.df.empty:
             return self.df._default_to_pandas(lambda df: df.loc[key])
 
+        # Limited support for slicing operations on loc, and predominantly
+        # only with positional mapping by translating it to an iloc. To
+        # support proper labels (sorted) we will need to translate the
+        # existing index to a mask OR translate the expression to a set of
+        # predicates.
+        if is_slice(key):
+            if key.step is not None:
+                raise NotImplementedError("Slice with step not supported for loc")
+            if key.start is None and key.stop is None:
+                return self.df 
+            if key.start is None:
+                new_index = (self.df.index <= key.stop)
```

Comment:
This is interesting - does this logic match pandas behavior?

Reply:
Not exactly, in pandas this operation will produce a binary mask array, the same length as the array. We cannot quite do that here because the array would be the same size as the dataset.

Reply:
Once way to emulate this behavior is to provide a compressed "mask" index. Rather than materializing the actual booleans we would return a pandas.arrays.SparseArray